chutney/server/src/main/java/com/chutneytesting/execution/domain/PurgeServiceImpl.java

```diff
@@ -95,13 +116,13 @@ private static PurgeExecutionService<TestCaseMetadata, String, ExecutionSummary>
         );
     }
 
-    private PurgeExecutionService<Campaign, Long, CampaignExecution> buildCampaignService(CampaignRepository campaignRepository, CampaignExecutionRepository campaignExecutionRepository, Integer maxCampaignExecutions) {
+    private PurgeExecutionService<Campaign, Long, CampaignExecution> buildCampaignService(CampaignRepository campaignRepository, CampaignExecutionRepository campaignExecutionRepository, Integer maxCampaignExecutions, LocalDateTime maxCampaignExecutionsDateConfiguration) {
         return new PurgeExecutionService<>(
             maxCampaignExecutions,
             campaignRepository::findAll,
             campaign -> campaign.id,
             campaignExecutionRepository::getExecutionHistory,
-            cer -> true,
+            cer ->  Optional.ofNullable(maxCampaignExecutionsDateConfiguration).map(cer.startDate::isBefore).orElse(true),
```

Comment:
I think it's not right. The date to compare cannot be fixed but re-calculated at each purge.
I'll try to fix this.